seed/static/seed/js/controllers/inventory_list_controller.js

```diff
@@ -111,6 +118,201 @@ angular.module('BE.seed.controller.inventory_list', [])
+      $scope.isModified = function () {
+        current_filters = {};
+        current_filters = inventory_service.get_format_column_filters($scope.column_filters);
+        saved_filters = $scope.currentFilterGroup.query_dict;
+        current_labels = [];
+        for (label in $scope.selected_labels) {
+          current_labels.push($scope.selected_labels[label].id);
+        };
+        saved_labels = $scope.currentFilterGroup.labels;
+        current_label_logic = $scope.labelLogic;
+        saved_label_logic = $scope.currentFilterGroup.label_logic;
+        if (!_.isEqual(current_filters, saved_filters)) {
+          $scope.Modified = true
+          console.log("filters are different");
+        } else if (!_.isEqual(current_labels.sort(), saved_labels.sort())) {
+          $scope.Modified = true
+          console.log("labels are different");
+        } else if (current_label_logic !== saved_label_logic) {
+          $scope.Modified = true
+          console.log("label_logic is different");
+        } else {
+          $scope.Modified = false
+          console.log("no changes");
+        }
+        return $scope.Modified;
+      };
+
+      $scope.check_for_changes = function () {
+        if ($scope.Modified) {
+          $uibModal.open({
+            template: '<div class="modal-header"><h3 class="modal-title" translate>You have unsaved changes</h3></div><div class="modal-body" translate>You will lose your unsaved changes if you switch filter groups without saving. Would you like to continue?</div><div class="modal-footer"><button type="button" class="btn btn-warning" ng-click="$dismiss()" translate>Cancel</button><button type="button" class="btn btn-primary" ng-click="$close()" autofocus translate>Switch Filter Groups</button></div>'
+          }).result.then(function () {
+            $scope.Modified = false;
+          }).catch(function () {
+            return;
+          });
+        }
+
+        updateCurrentFilterGroup($scope.currentFilterGroup)
+      };
+
+      updateCurrentFilterGroup = (filterGroup) => {
+        // Set current filter group
+        $scope.currentFilterGroup = filterGroup;
+        filter_groups_service.save_last_filter_group($scope.currentFilterGroup.id, $scope.inventory_type)
+
+        // Update labels
+        $scope.labelLogicUpdated($scope.currentFilterGroup.label_logic);
+        $scope.selected_labels = _.filter($scope.labels, function (label) {
+          return _.includes($scope.currentFilterGroup.labels, label.id);
+        });
+
+        // clear table filters
+        $scope.gridApi.grid.columns.forEach(column =>{
+          column.filters[0] = {
+            term: null
+          };
+        })
+
+        // write new filter in table
+        for (const key in $scope.currentFilterGroup["query_dict"]) {
+          const value = $scope.currentFilterGroup["query_dict"][key]
```

Comment:
Just a suggestion, by iterating with `Object.entries() of` you can define both `key` and `value` as iterable variables
```suggestion
        for (const [key, value] of Object.entries($scope.currentFilterGroup["query_dict"])) {
```